src/func.d

```diff
@@ -4018,6 +3991,36 @@ extern (C++) class FuncDeclaration : Declaration
         return fd;
     }
 
+    /******************
+     * Check parameters and return type of D main() function.
+     * Issue error messages.
+     */
+    final void checkDmain()
+    {
+        TypeFunction tf = cast(TypeFunction)type;
+        const nparams = Parameter.dim(tf.parameters);
+        bool argerr;
+        if (nparams == 1)
+        {
+            auto fparam0 = Parameter.getNth(tf.parameters, 0);
+            auto t = fparam0.type.toBasetype();
+            if (t.ty != Tarray ||
+                t.nextOf().ty != Tarray ||
+                t.nextOf().nextOf().ty != Tchar ||
+                fparam0.storageClass & (STCout | STCref | STClazy))
+            {
+                argerr = true;
```

Comment:
So it's behavior remained the same as the previous code w.r.t. what errors are printed for each sort of error.